lib/rubocop/cop/lint/use_positive_int32_validator.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+require 'byebug'
+
+module RuboCop
+  module Cop
+    module Lint
+      # @example
+      #   # bad
+      #   requires :id, type: Integer, desc: 'Comment ID'
+      #
+      #   # good
+      #   requires :id, type: Integer, desc: 'Comment ID', positive_int32: true
+      class UserPositiveInt32Validator < Cop
+        MSG = 'If this Integer maps to a postgres Integer column, validate with `positive_int32: true`'.freeze
+
+        def_node_matcher :get_require_with_integer_hash, <<~PATTERN
+        (send nil? :requires
+          (sym _)
+          $(hash ...) ...)
+        PATTERN
+        def_node_search :is_type_integer?, '(pair (sym :type) (const nil? :Integer))'
+        def_node_search :validates_integer?, '(pair (sym :positive_int32) (true))'
```

Comment:
It took me way too long to figure out why the parser chokes on `(hash ... (pair ...) ...)`. So I used `def_node_search` instead of `def_node_matcher` to check the hash's pairs